.github/workflows/haskell.yml

```diff
@@ -4,20 +4,26 @@ on:
   push:
     branches: [ "master" ]
   pull_request:
+  merge_group:
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
+
 
 permissions:
   contents: read
 
 jobs:
+  # Build and test
   build:
-
     runs-on: ${{ matrix.os }}
 
     strategy:
       fail-fast: false
       matrix:
-        ghc: ["8.10.7", "9.2.5"]
-        cabal: ["3.8.1.0"]
+        ghc: ["8.10.7", "9.2.8", "9.4.8", "9.6.4", "9.8.2"]
+        cabal: ["3.10.3.0"]
         os: [ubuntu-latest, windows-latest, macOS-latest]
```

Comment:
I'm now thinking that we might actually want to use the `sys {os, shell}` approach that was there in lmdb-simple already.

Reply:
Changed in 068d068c9632295e66512b2176fa54c2022455d0. Can I squash and merge like this?